File: ios/Classes/FileEncrypterPlugin.m
#import "FileEncrypterPlugin.h"
#if __has_include(<file_encrypter/file_encrypter-Swift.h>)
#import <file_encrypter/file_encrypter-Swift.h>
#else
#import "file_encrypter-Swift.h"
#endif

@implementation FileEncrypterPlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftFileEncrypterPlugin registerWithRegistrar:registrar];
}
@end

File: ios/Classes/FileEncrypterPlugin.swift
import Flutter
import UIKit

public class FileEncrypterPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "file_encrypter", binaryMessenger: registrar.messenger())
    let instance = FileEncrypterPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
